src/libraries/icubmod/rawValuesPublisherClient/RawValuesPublisherClient.cpp

```diff
@@ -0,0 +1,192 @@
+/*
+ * SPDX-FileCopyrightText: 2006-2024 Istituto Italiano di Tecnologia (IIT)
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "RawValuesPublisherClient.h"
+
+#include "EoProtocol.h"
+#include "EoProtocolMC.h"
+#include <yarp/os/LogComponent.h>
+
+namespace {
+    YARP_LOG_COMPONENT(RAWVALUESPUBLISHERCLIENT, "iCub.debugLibrary.rawvaluespublisherclient")
+}
+
+#undef DEBUG_RAW_VALUES_MACRO 
+
+using namespace yarp::os;
+
+// Static inline functions
+static inline bool NOT_YET_IMPLEMENTED(const char* txt)
+{
+    yCError(RAWVALUESPUBLISHERCLIENT) << txt << "has not yet been emplemented";
+    return true;
+}
+
+static inline bool DEPRECATED(const char* txt)
+{
+    yCError(RAWVALUESPUBLISHERCLIENT) << txt << "has been deprecated";
+    return true;
+}
+
+void RawValuesStreamingDataInputPort::onRead(rawValuesDataVectorsMap& rawdata)
+{
+    // saving read data from server output port in the local map
+    std::lock_guard<std::mutex> guard(dataMutex);
+    receivedRawDataMap = rawdata.vectorsMap;
+    lastTimeStampReadInSeconds = yarp::os::Time::now();
+}
+
+
+void RawValuesStreamingDataInputPort::updateTimeoutStatus() const
+{
```

Comment:
I'll remove it, since I saw that is never called. I took inspiration by the MultipleAnalogSensorClient but I've realized that in our case is never triggered. I thought it was automatically called as per open() and attach() methods.